src/Workspaces/Core/Portable/Options/EditorConfigStorageLocation.cs

```diff
@@ -75,11 +77,19 @@ public EditorConfigStorageLocation(string keyName, Func<string, object> parseVal
         public EditorConfigStorageLocation()
         {
             // If the user didn't pass a keyName assume we need to parse the entire dictionary
-            _parseDictionary = (dictionary, type) =>
+            _tryParseDictionary = (dictionary, type) =>
             {
                 if (type == typeof(NamingStylePreferences))
                 {
-                    return EditorConfigNamingStyleParser.GetNamingStylesFromDictionary(dictionary);
+                    var result = EditorConfigNamingStyleParser.GetNamingStylesFromDictionary(dictionary);
+                    if (!result.NamingRules.Any() &&
+                        !result.NamingStyles.Any() &&
+                        !result.SymbolSpecifications.Any())
```

Comment:
This raises an interesting question about how naming styles should cascade.  As soon as there is one from an editorconfig file, that removes any consideration of VS options?  Seems reasonable, but tagging @kuhlenh for thoughts too.